### ISSUE TITLE
sensor naming scheme has been changed with HA 2023.11

### ISSUE THREAD

**Issue by FauthD** [FL550/dwd_weather#109]
Just to report this.

Not sure if it was caused by an update of this integration or by HA 2023.9 -> 2023.11 (I did not use 2023.10).

The name used to be "sensor.sun_irradiance_schorndorf".
Now after the the upgrade to 2023.11 it is:
"sensor.schorndorf_sun_irradiance".

Is this a known change?
I turned on the debug logs now, can post it later.

Thanks, Dieter

**Comment by FauthD** [comment]
Hi, 
might be on purpose starting with V2.0.
In sensor.py line 201 ff:
```
class DWDWeatherForecastSensor(DWDWeatherEntity, SensorEntity)
...
        name = f"{dwd_data.dwd_weather.station_name}: {SENSOR_TYPES[self._type][0]}"
        unique_id = f"{dwd_data.dwd_weather.station_id}_{SENSOR_TYPES[self._type][0]}"
...
```
The order was vice versa in V1.xx.

However I did not find and hint about that in the docs. Tomatoes on my eyes? :-)

The sensor.schorndorf_sun_irradiance contains hourly values, but all values are **None**.

The log was extracted from home-assistant.log with `grep dwd_weather`

[dwd_weather.log](https://github.com/FL550/dwd_weather/files/13393995/dwd_weather.log)

**Comment by FL550** [comment]
> The name used to be "sensor.sun_irradiance_schorndorf".
> Now after the the upgrade to 2023.11 it is:
> "sensor.schorndorf_sun_irradiance".

Yes, this had to be done to avoid errors with duplicate names. Sorry for any inconveniences.

**Comment by FL550** [comment]
> The sensor.schorndorf_sun_irradiance contains hourly values, but all values are None.

The values are not provided by DWD. See the raw file here: https://opendata.dwd.de/weather/local_forecasts/mos/MOSMIX_L/single_stations/Q375/kml/MOSMIX_L_LATEST_Q375.kmz

**Comment by FL550** [comment]
> BTW: What was the reason to recalc the irradiance values (/ 3.6) back in may 2023?
> Had broken my roller blinds shading algo.

Please see the release notes: https://github.com/FL550/dwd_weather/releases/tag/v1.2.26

**Comment by FauthD** [comment]
Hi FL550,
thanks for the answers and no problem, just wanted to understand (/ 3.6).

What I wonder is that irradiation values showed up in V1.x a few days ago. Perhaps I missed the disapearence? I use it for my roller blinds and these open completely in the morning these days, so I probably did not find out.

In [geoportal](https://dwd-geoportal.de/products/G_FJM/) I found a link that shall dexribe the kmz format, but is 404 now.
Is there some other info I can use to understand the kmz format? Or a tool to read it?

The irradiation was a good input to my blinds algoritm. They worked perfect last summer. (is an AppDaemon app)

Is there a chance the irradiation is contained in other kmz files (not too far away)?

Thanks for you work on this integration!
Dieter

**Comment by FauthD** [comment]
Is this the same as the geo infomation format kmz/kml?
Looks similar. from naming.
Could unzip the kmz into kml and now see the text.
Hopefully I find a nearby source for the irradiation.

**Comment by FauthD** [comment]
[kml/kmz Info](https://www.dwd.de/DE/leistungen/opendata/neuigkeiten/opendata_mar2018_01.html)

**Comment by FauthD** [comment]
Found a station that supports the irradiation.

I had used Station Q375: Winterbach/Remstal based on automatic selection.
But there is an even closer station: "P0249: Schorndorf" and that supports irradiation.
I also checked other stations in my area (Schwäbisch Gmünd, WInterbach). Also no support for irradiation.
As long as P0249 provides irradiation data, I will be happy.
 
Thanks again, you steared me in the right direction.
Dieter
